agent/php_minit.c

```diff
@@ -38,6 +38,24 @@
 #include "fw_laravel.h"
 #include "lib_guzzle4.h"
 
+#include "php_execute.h"
+
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP8+ */
+/*
+ * Register the begin and end function handlers with the Observer API.
+ */
+static zend_observer_fcall_handlers nr_php_fcall_register_handlers(
+    zend_execute_data* execute_data) {
+  zend_observer_fcall_handlers handlers = {NULL, NULL};
+  if (!execute_data->func || !execute_data->func->common.function_name) {
```

Comment:
Are we sure these checks are needed? Are these values set when using PHP CLI SAPI on a file with no functions in it? Consider following input file `simple-script.php`:
```
<?php
echo "Hello world!\n"
```
`execute_data->func !=nil` but `execute_data->func->common.function_name==nil` for this example. Maybe this PR is not the place for changing this and it is scheduled for later work.

Reply:
We would still want to instrument in that case, so the first clause would let through, and it would just be unnamed (basically instrumenting a file).  We want to instrument when it is a file or a function.

For instance, in this test case, you can see the transaction/segment was named after file because there was no function name.

We want to observe functions and files(this is the case that you mentioned above where a function isn't explicitly named).  PHP does something similar in their example: https://github.com/php/php-src/blob/master/ext/zend_test/observer.c#L184

Reply:
Addressed in 4f2b2a618c781fa0db66796ad12423a4e63607b5 and 4f2b2a618c781fa0db66796ad12423a4e63607b5.